pallets/marketplace/src/lib.rs

```diff
@@ -251,49 +438,67 @@ pub mod pallet {
             registration: &JobRegistrationFor<T>,
         ) -> Result<(), DispatchError> {
             let e: <T as Config>::RegistrationExtra = registration.extra.clone().into();
-            let extra: JobRequirementsFor<T> = e.into();
+            let requirements: JobRequirementsFor<T> = e.into();
 
             ensure!(
-                extra.cpu_milliseconds > 0,
-                Error::<T>::JobRegistrationZeroCPUMilliseconds
+                registration.schedule.duration > 0,
+                Error::<T>::JobRegistrationZeroDuration
+            );
+            ensure!(
+                registration.schedule.execution_count() <= MAX_EXECUTIONS_PER_JOB,
+                Error::<T>::JobRegistrationZeroDuration
```

Comment:
This error is misleading. It should be something like `JobRegistrationTooManyExecutions`